plenum/test/replica/test_ordered_tracker.py

```diff
@@ -0,0 +1,109 @@
+from random import randint
+
+import pytest
+from plenum.server.replica import IntervalList, OrderedTracker
+
+
+@pytest.fixture
+def intervals():
+    return IntervalList()
+
+
+@pytest.fixture
+def tracker():
+    return OrderedTracker()
+
+
+def test_interval_list_has_contains_accessor(intervals: IntervalList):
+    assert 1 not in intervals
+
+
+def test_interval_list_can_add_value(intervals: IntervalList):
+    intervals.add(1)
+
+    assert 0 not in intervals
+    assert 1 in intervals
+    assert 2 not in intervals
+
+
+def test_interval_list_can_add_continuous_blocks(intervals: IntervalList):
+    intervals.add(1)
+    intervals.add(2)
+
+    assert 0 not in intervals
+    assert 1 in intervals
+    assert 2 in intervals
+    assert 3 not in intervals
+
+
+def test_interval_list_can_grow_blocks_in_arbitrary_direction(intervals: IntervalList):
+    intervals.add(1)
+    intervals.add(2)
+    intervals.add(0)
+
+    assert -1 not in intervals
+    assert 0 in intervals
+    assert 1 in intervals
+    assert 2 in intervals
+    assert 3 not in intervals
+
+
+def test_interval_list_can_have_multiple_blocks(intervals: IntervalList):
```

Comment:
Please add more tests:
- add the same value multiple times
- add more tests for adding in different order, for example: 
  -- 2,1,4,3 
  -- 1,2,4,5,6,3

Reply:
Actually such cases are covered by a bit higher level randomized tests of `OrderedTracker`

Reply:
Also, it might be beneficial to employ Hypothesis framework (https://hypothesis.readthedocs.io/en/latest/) here.